File: flutter/feature_core/lib/src/ui/splash/splash_screen.dart
import 'package:flutter/material.dart';
import 'package:get_it/get_it.dart';
import 'package:contracts/contracts.dart';
import 'package:go_router/go_router.dart';

/// Splash screen — restores session on launch then navigates.
///
/// Navigates to [Routes.home] if a session is restored, [Routes.login] otherwise.
/// Mirrors Android [SplashScreen].
class SplashScreen extends StatefulWidget {
  const SplashScreen({super.key});

  @override
  State<SplashScreen> createState() => _SplashScreenState();
}

class _SplashScreenState extends State<SplashScreen> {
  @override
  void initState() {
    super.initState();
    _checkSession();
  }

  Future<void> _checkSession() async {
    // Brief pause for splash visibility.
    await Future<void>.delayed(const Duration(milliseconds: 500));
    if (!mounted) return;

    final authService = GetIt.instance<AuthService>();
    final user = await authService.restoreSession();
    if (!mounted) return;

    context.go(user != null ? Routes.home : Routes.login);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Text(
              'LargeScale',
              style: Theme.of(context).textTheme.headlineLarge?.copyWith(
                    color: Theme.of(context).colorScheme.primary,
                  ),
            ),
            const SizedBox(height: 8),
            Text(
              'Modular · Multi-tenant',
              style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                    color: Theme.of(context).colorScheme.onSurfaceVariant,
                  ),
            ),
            const SizedBox(height: 32),
            const CircularProgressIndicator(),
          ],
        ),
      ),
    );
  }
}

File: android/feature-core/src/main/java/com/densitech/largescale/feature/core/ui/splash/SplashScreen.kt
package com.densitech.largescale.feature.core.ui.splash

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.densitech.largescale.contracts.AuthService
import com.densitech.largescale.contracts.Routes
import com.densitech.largescale.shared.ui.components.LoadingIndicator
import com.densitech.largescale.shared.ui.theme.AppTheme
import com.densitech.largescale.shared.ui.theme.Spacing
import kotlinx.coroutines.delay

/**
 * Entry point screen. Attempts to restore a saved session.
 * Navigates to HOME on success, LOGIN on failure.
 *
 * @param authService   Injected at the call site (from `:app` or NavGraph setup)
 * @param onNavigate    Lambda accepting the target route — caller drives navigation
 */
@Composable
fun SplashScreen(
    authService: AuthService,
    onNavigate: (route: String) -> Unit,
    modifier: Modifier = Modifier
) {
    LaunchedEffect(Unit) {
        delay(500) // Brief pause for splash visibility
        val user = authService.restoreSession()
        onNavigate(if (user != null) Routes.HOME else Routes.LOGIN)
    }

    Box(modifier = modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text(
                text = "LargeScale",
                style = MaterialTheme.typography.headlineLarge,
                color = MaterialTheme.colorScheme.primary
            )
            Spacer(Modifier.height(Spacing.sm))
            Text(
                text = "Modular · Multi-tenant",
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
            Spacer(Modifier.height(Spacing.lg))
            LoadingIndicator(modifier = Modifier.height(Spacing.xxl))
        }
    }
}

@Preview(showBackground = true)
@Composable
private fun SplashScreenPreview() {
    AppTheme {
        Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text("LargeScale", style = MaterialTheme.typography.headlineLarge,
                    color = MaterialTheme.colorScheme.primary)
                Spacer(Modifier.height(Spacing.sm))
                Text("Modular · Multi-tenant", style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant)
            }
        }
    }
}
